pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -345,6 +335,28 @@ if [[ -n $flake ]]; then
     fi
 fi
 
+tmpDir=$(mktemp -t -d nixos-rebuild.XXXXXX)
```

Comment:
Also move `cleanup()` and `trap` here. Otherwise the temporary directory won't be removed on reexec.